### PR TITLE
ECIL-216 Check payload goods description for invalid whitespace characters

### PR DESCRIPTION
**Pull request by mikeylockwood**

### CODE REVIEW

File: mail/serializers.py [hunk 1]
```diff
@@ -198,3 +205,9 @@ def _validate_controlled_by(data):
 
         if not data.get("unit"):
             raise serializers.ValidationError("'unit' must be set when controlled_by equals 'Q'")
+
+
+def _validate_description_whitspace(description):
```
Comment (MattHolmes123) [inline mail/serializers.py:210]: Can we type this please.

File: mail/serializers.py [hunk 2]
```diff
@@ -198,3 +205,9 @@ def _validate_controlled_by(data):
...
+    """Check for invalid whitespace characters in description"""
+    if len(description.splitlines()) > 1 or "\t" in description:
+        raise serializers.ValidationError("'description' contains an invalid whitespace character")
```
Comment (MattHolmes123) [inline mail/serializers.py:213]:
```
>>> "asdf\n".splitlines()
['asdf']
>>> print("asdf\n")
asdf

```

Comment (MattHolmes123) [reply mail/serializers.py:213]: This won't raise an error if the string has a single newline at the end

Comment (mikeylockwood) [reply mail/serializers.py:213]: Django forms will automatically strip the whitespace characters from the ends of the string. Maybe there could have been trailing characters from V1 data though

**Review by MattHolmes123 — APPROVED** [review]
Will leave it with you.
Feel free to fix the validator for trailing new lines (just incase) as that will cause issues if it gets through.
Same with using field level validation functions if you so desire 👍

File: mail/serializers.py (hunk 2 above)
Comment (mikeylockwood) [reply mail/serializers.py]: [CharField](https://www.django-rest-framework.org/api-guide/fields/#charfield) in DRF serializers also trims the whitespace. I have changed the check anyway to include the end characters

**Pull request merged**